src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CounterPayloadExtensions.cs

```diff
@@ -1,21 +1,46 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
+
 namespace Microsoft.Diagnostics.Monitoring.EventPipe
 {
     internal static class CounterPayloadExtensions
     {
-        public static string GetDisplay(this ICounterPayload counterPayload)
+        internal enum DisplayRenderingMode
+        {
+            Default,
+            DotnetCounters
+        }
+
+        // It appears that every instance where this is called is using DisplayRenderingMode.DotnetCounters - do we want to simplify this method to make that assumption?
+        public static string GetDisplay(this ICounterPayload counterPayload, DisplayRenderingMode displayRenderingMode = DisplayRenderingMode.Default)
         {
-            if (counterPayload.CounterType == CounterType.Rate)
+            if (!counterPayload.IsMeter)
             {
-                return $"{counterPayload.DisplayName} ({counterPayload.Unit} / {counterPayload.Interval} sec)";
+                if (counterPayload.CounterType == CounterType.Rate)
+                {
+                    return $"{counterPayload.DisplayName} ({GetUnit(counterPayload.Unit, displayRenderingMode)} / {GetInterval(counterPayload, displayRenderingMode)} sec)";
+                }
+                if (!string.IsNullOrEmpty(counterPayload.Unit))
+                {
+                    return $"{counterPayload.DisplayName} ({counterPayload.Unit})";
+                }
             }
-            if (!string.IsNullOrEmpty(counterPayload.Unit))
+
+            return $"{counterPayload.DisplayName}";
+        }
+
+        private static string GetUnit(string unit, DisplayRenderingMode displayRenderingMode)
+        {
+            if (displayRenderingMode == DisplayRenderingMode.DotnetCounters && string.Equals(unit, "count", StringComparison.OrdinalIgnoreCase))
```

Comment:
nit: Make some consts for the literals here.

Reply:
This is a rare case where I think that might slightly obscure readability since the strings explain themselves better than named constants and they're only used once - currently planning to leave as-is.